experimental/form-engine/USER_GUIDE.md

```diff
@@ -0,0 +1,1003 @@
+   ```jsx
+   <Form
+     validate={(values) => {
+       // Return object for field-specific errors
+       if (values.email?.endsWith('@banned.com')) {
+         return { email: 'This domain is not allowed' };
+       }
+       // Return string for form-level error
+       if (values.items?.length === 0) {
+         return 'At least one item is required';
+       }
+       return null;
+     }}
+   />
+   ```
+
+3. **Error strategies** handle different formats automatically:
+   - `NullErrorStrategy` - Null/undefined (no error)
+   - `ObjectErrorStrategy` - `{ field: 'error' }` format
+   - `ArrayErrorStrategy` - `['error1', 'error2']` format (logs warning)
+   - `StringErrorStrategy` - `'error message'` format (form-level)
+
+4. **Accessing errors**:
+   - `meta.error` - First error message
+   - `meta.errors` - All errors with sources: `[{ source: 'field'|'form', error: string }]`
+
+5. **Error priority**:
+   - Both field-level and form-level errors are stored
+   - Form-level errors can overwrite field-level errors
+   - Last write wins (no merging)
+
+### Hooks
+
+**useField(name, options)**
+
+Subscribe to specific field state.
+
+```jsx
+const { input, meta } = useField('email', {
+  validate: (value) => !value ? 'Required' : undefined,
+  validateOn: 'blur',
+  debounceDelay: 300,
+  subscription: { value: true, error: true },
+});
+```
+
+**useFormState(subscription)**
+
+Subscribe to form-level state.
+
+```jsx
+const formState = useFormState({
+  values: true,      // All form values
+  errors: true,      // All field errors
+  touched: true,     // Touched fields
+  active: true,      // Currently active field
+  submitting: true,  // Submission in progress
+  submitSucceeded: true, // Last submission succeeded
+  dirty: true,       // Form has unsaved changes
+  pristine: true,    // Form has no unsaved changes
+  valid: true,       // All validations passed
+});
+```
+
+**useWatch(name, options)**
+
+Watch specific field with advanced features.
+
+```jsx
+// Basic watch
+const email = useWatch('email');
+
+// With selector (transform value)
+const emailLength = useWatch('email', {
+  selector: (value) => value?.length || 0,
+});
+
+// With bubble (receive nested field changes)
+const orders = useWatch('orders', { bubble: true });
+// Updates when orders[0].amount, orders[1].status, etc. change
+
+// Combine selector and bubble
+const orderCount = useWatch('orders', {
+  selector: (arr) => arr?.length || 0,
+  bubble: true,
+});
+```
+
+**Bubble Behavior:**
+- `bubble: false` (default) - Only updates when exact field changes
+- `bubble: true` - Updates when any nested field changes
+  - Example: `useWatch('foo', { bubble: true })` receives events from `foo[0].field`, `foo.bar`, etc.
+  - Useful for watching arrays/objects and updating when ANY nested value changes
+  - Performance optimized - only emits to bubble listeners when nested fields change
+
+**useFormSubmit(onSubmit)**
```

Comment:
Field-related hooks do not include `Field` in their names (e.g. it's `useWatch` not `useWatchField`) so be consistent and just call this `useSubmit`.